docs/designs/xref.md

```diff
@@ -4,6 +4,143 @@ Besides using file path to link to another file, DocFX also allows you to give a
 - API reference documentation is usually auto generated so it's difficult to find its file path.
 - You want to reference to files in another project without need to know its file structure.
 
+## Feature Requirements/Scenarios
+- Define an internal UID and reference to this UID within the current repository
+  ```yaml
+  inputs:
+    docs/a.md: |
+    ---
+    title: Title from yaml header a
+    uid: a
+    ---
+    docs/b.md: Link to @a
+  outputs:
+    docs/b.json: |
+        {"conceptual":"<p>Link to <a href=\"a\">Title from yaml header a</a></p>\n"}
+  ```
+- Define multiple UID with same value internally with different versions and reference to this UID without versioning within the current repository
+    - If all versions for this UID are within the same product, take the one with highest versioning respecting the referencing file
+      ```yaml
+        # v1: 1.0, 2.0, 3.0
+        # v2: 4.0, 5.0
+        # v3: 3.0 
+        # should take the highest version and respect the version of the referencing file
+        inputs:
+            docs/v1/a.md: |
+            ---
+            title: Title from v1
+            uid: a
+            ---
+            docs/v2/a.md: |
+            ---
+            title: Title from v2
+            uid: a
+            ---
+            docs/v3/b.md: Link to @a
+        outputs:
+            docs/b.json: |
+                {"conceptual":"<p>Link to <a href=\"a\">Title from v1</a></p>\n"}
+        ```
+        ```yaml
+        # v1: 1.0, 2.0, 3.0
+        # v2: 4.0, 5.0
+        # should take the highest version
+        inputs:
+            docs/v1/a.md: |
+            ---
+            title: Title from v1
+            uid: a
+            ---
+            docs/v2/a.md: |
+            ---
+            title: Title from v2
+            uid: a
+            ---
+            docs/b.md: Link to @a
+        outputs:
+            docs/b.json: |
+                {"conceptual":"<p>Link to <a href=\"a\">Title from v2</a></p>\n"}
+        ```
+    - If the versions for this UID are from different products, take the highest one base on proudct name alphabetically
+        ```yaml
+        # v1: a-1.0, a-2.0, a-3.0
+        # v2: b-4.0, b-5.0
+        # v3: c-1.0
+        # should take the hi
+        inputs:
+            docs/v1/a.md: |
+            ---
+            title: Title from v1
+            uid: a
+            ---
+            docs/v2/a.md: |
+            ---
+            title: Title from v2
+            uid: a
+            ---
+            docs/b.md: Link to @a
+        outputs:
+            docs/b.json: |
+                {"conceptual":"<p>Link to <a href=\"a\">Title from v2</a></p>\n"}
+        ```
+- Reference to an external UID without versioning
```

Comment:
Its great to list the status qoo, I would be more interested in why things are implemented the way it is, e.g. because of some limitations in the hosting layer. Bottom line is we implement these as is and later rework the whole fallback story with hosting together